python/src/trezorlib/cli/ethereum.py

```diff
@@ -164,14 +167,96 @@ def _format_access_list(
     ]
 
 
+def _hex_or_file(data: str) -> bytes:
+    try:
+        path = Path(data)
+        if path.is_file():
+            return path.read_bytes()
+    except Exception:
+        pass
+
+    try:
+        if data.startswith("0x"):
+            data = data[2:]
+        return bytes.fromhex(data)
+    except Exception as e:
+        raise click.ClickException(f"Invalid hex or file path: {data}") from e
+
+
+class CliSource(definitions.Source):
+    network: Optional[bytes] = None
+    token: Optional[bytes] = None
+    delegate: definitions.Source = definitions.NullSource()
+
+    def get_network(self, chain_id: int) -> Optional[bytes]:
+        if self.network is not None:
+            return self.network
+        return self.delegate.get_network(chain_id)
+
+    def get_network_by_slip44(self, slip44: int) -> Optional[bytes]:
+        if self.network is not None:
+            return self.network
+        return self.delegate.get_network_by_slip44(slip44)
+
+    def get_token(self, chain_id: int, address: Any) -> Optional[bytes]:
+        if self.token is not None:
+            return self.token
+        return self.delegate.get_token(chain_id, address)
+
+
+DEFINITIONS_SOURCE = CliSource()
+
+
 #####################
 #
 # commands start here
 
 
 @click.group(name="ethereum")
-def cli() -> None:
+@click.option(
+    "-d", "--definitions", "defs", help="Source for Ethereum definition blobs."
+)
+@click.option(
+    "-a",
+    "--auto-definitions",
+    is_flag=True,
+    help="Automatically download required definitions from trezor.io",
+)
+@click.option("--network", help="Network definition blob.")
+@click.option("--token", help="Token definition blob.")
+def cli(
+    defs: Optional[str],
+    auto_definitions: Optional[bool],
+    network: Optional[str],
+    token: Optional[str],
+) -> None:
     """Ethereum commands."""
+    if auto_definitions:
+        if defs is not None:
+            raise click.ClickException(
+                "Cannot use --definitions and --auto-definitions at the same time."
+            )
+        DEFINITIONS_SOURCE.delegate = definitions.UrlSource()
+    elif defs is not None:
+        try:
+            path = Path(defs)
+            if path.is_dir():
+                DEFINITIONS_SOURCE.delegate = definitions.FilesystemSource(path)
+            elif path.is_file() and tarfile.is_tarfile(path):
+                DEFINITIONS_SOURCE.delegate = definitions.TarSource(path)
+        except Exception:
```

Comment:
Same as above, probably redundant `except`, nothing should be raised

The whole block could theoretically be

```python
path = Path(defs)
if path.is_dir():
    DEFINITIONS_SOURCE.delegate = definitions.FilesystemSource(path)
elif path.is_file() and tarfile.is_tarfile(path):
    DEFINITIONS_SOURCE.delegate = definitions.TarSource(path)
elif defs.startswith("http"):
    DEFINITIONS_SOURCE.delegate = definitions.UrlSource(defs)
else:
    raise click.ClickException("Unrecognized --definitions value.")
```